### PR TITLE
[insurance-beta] savecart module with insurance before release

### PR DESCRIPTION
**Pull request by Benjamin-Freoua-Alma**
### Reason for change

[Linear task](https://linear.app/almapay/issue/ECOM-1691/[🧩-ps]-fix-compatibility-with-a-savecart-module)

### Code changes

Add specific code for compatibility with the module opartSaveCart and Insurance

### How to test

_As a reviewer, you are encouraged to test the PR locally._
Installe the module OpartSaveCart
Save the cart
Get the cart saved
Change your current cart
And re-get your cart saved

### Checklist for authors and reviewers

- [ ] The title of the PR uses business wording, not technical jargon, for the changelog readers to understand it
- [ ] The PR implements the changes asked in the referenced task / issue
- [ ] The automated tests are compliant with the [testing strategy](https://www.notion.so/almapay/Backend-testing-strategy-06c642cec1bf47b9b8feca3a91ea8d4a)
- [ ] The tests are relevant, and cover the corner/error cases, not only the happy path
- [ ] You understand the impact of this PR on existing code/features
- [ ] The changes include adequate logging and Datadog traces
- [ ] Documentation is updated (API, developer documentation, ADR, Notion...)

### CODE REVIEW

File: alma/lib/Repositories/AlmaInsuranceProductRepository.php [hunk 1]
```diff
@@ -190,6 +190,24 @@ public function getByOrderId($id)
...
 
+    /**
+     * @param $cartId
+     * @param $shopId
+     *
+     * @return mixed
+     *
+     * @throws \PrestaShopDatabaseException
+     */
+    public function getByCartIdAndShop($cartId, $shopId)
```
Comment (Francois-Gomis) [inline alma/lib/Repositories/AlmaInsuranceProductRepository.php:201]: get what ?

File: alma/controllers/hook/ActionCartSaveHookController.php [hunk 2]
```diff
@@ -81,10 +87,27 @@ public function __construct($module)
...
      */
     public function run($params)
     {
+        $currentCart = $this->context->cart;
+        $newCart = $params['cart'];
+
+        // TODO : Need to optimise for more that the module opartSaveCart
+        if (null === $currentCart->id || $currentCart->id != $newCart->id) {
+            if (\Tools::getValue('action') !== 'shareCart') {
+                $cartIdSaved = $this->cartSaveService->getCartSaved(\Tools::getValue('token'));
```
Comment (Francois-Gomis) [inline alma/controllers/hook/ActionCartSaveHookController.php:100]: getCartSaved can be false

File: alma/lib/Services/CartSaveService.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        $this->moduleManagerBuilder = $moduleManagerBuilder;
+        $this->cartSaveRepository = $cartSaveRepository;
+    }
+
+    /**
+     * @param $value
+     *
+     * @return array|bool
+     */
+    public function getCartSaved($value)
```
Comment (Francois-Gomis) [inline alma/lib/Services/CartSaveService.php:65]: Not explicit name

**Comment by Benjamin-Freoua-Alma** [comment]
@Francois-Gomis We make the fix of your review in synchronous. Thanks

**Review by ClaireAlma — APPROVED** [review]

**Review by Francois-Gomis — APPROVED** [review]

**Pull request merged**